rsa-cryptosystem-signature/index08.js

```diff
@@ -0,0 +1,21 @@
+let publicKey = 0;
+
+function hashTheMessage(message) {
+  let hashValue = 0;
+  for (let i = 0, msgLength = message.length; i < msgLength; ++i) {
+    hashValue += message.charCodeAt(i);
+  }
+  return hashValue;
+}
+
+function generatePrivateKey() {}
+
+function generatePublicKey() {}
+
+/*
+Suppose Alice sends an encrypted message to Bob using a symmetric key. There's no way for Bob to verify if the sender is Alice. Since anyone with access to key can send Bob the message and claim to be Alice.
```

Comment:
It would probably be better to build out one function at a time here, then we could work on breaking up this explanation.

Just so I understand how this process works, I'll try to summarize the process below. Please let me know if I'm missing anything!:
1. Alice wants to send a message to Bob, so she hashes and and encrypts a message using Bob's private key. She sends the unencrypted message, along with the hashed and encrypted message otherwise known as a signature.
2. Bob decrypts the signature which results in the original hashed message. He also hashes the unecrypted message he received and compares the hashed value to the decrypted signature.
3. If both match, then he's sure that the message is from Alice since she's the only one with his private key.

Is that how this process works?

Reply:
Yes, exactly that's how it works!

Reply:
This explanation is the heart of the lesson. Campers must be able to grasp it properly. Should we improve the explanation? or it's fine?

Reply:
@vkWeb, messed up the steps originally, so I just edited them above. :point_up:

Yes, I agree. It's important that we nail this explanation, but it's quite a lot to take in all at once. It would probably be best if we can break it up among several lessons somehow.

Reply:
@scissorsneedfoodtoo We should strictly explain it in a single lesson because as we progress, we'll focus more on RSA.

Still, I'll keep my eyes open ;)

Reply:
Sure, just let me know. We're trying to keep lessons around 5 sentences or less, but there can be exceptions. The idea is that students will be working through these really fast, so everything should be as minimal as possible.

One idea I had is that you can explain parts of the process as students write the empty functions. Kind of a brief overview before you actually dive in and start coding. So you could have a lesson where students declare the empty `genereatePublicKey` function, and explain how each user should have a public key that can be used for encryption and decryption. Then the next lesson could be students declaring the empty `generatePrivateKey` function, and you explain that each user's public key is paired with a private key, and how they work together.

Perhaps you could continue the overview there, declaring empty functions around the editor and explaining how the whole process works before actually starting to code. What do you think?

Reply:
Yes, that empty function idea is my current approach of teaching. 

Umm. Okay, I'll break this explanation in parts and will see how things work.